modules/core/module.go

```diff
@@ -124,8 +124,8 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 	channeltypes.RegisterMsgServer(cfg.MsgServer(), am.keeper)
 	types.RegisterQueryService(cfg.QueryServer(), am.keeper)
 
-	m := clientkeeper.NewMigrator(am.keeper.ClientKeeper)
```

Comment:
Maybe this is due to my lack of knowledge of how migrations work, but if we replace `Migrate2to3` with `Migration3to4` and a chain upgrades from v6.0.0 to v7.1.0, wouldn't it miss the `Migrate2to3` migration?

Reply:
right, my understanding was that chains needed to update to each version one by one in order to run all the migrations. I feel like it should be possible, to just register all migrations at once (as they should be idempotent) however I'm not sure if this is the case. Maybe @AdityaSripal  has an answer to this?

Reply:
Yeah, I am also not sure if the chains really need to upgrade to each version of ibc-go or if they can jump versions as long as they run the migrations of the versions in between.

Reply:
if this is the case then we would preferably just add it as a migration instead of replace

Reply:
Yeah, it should be possible [to register multiple migrations](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/12-upgrade.md#registering-migrations).